refinery/tool_manager/models.py

```diff
@@ -556,17 +590,9 @@ def launch(self):
         """
         self._check_max_running_containers()
 
-        container = DockerContainerSpec(
-            image_name=self.tool_definition.image_name,
-            container_name=self.container_name,
-            labels={self.uuid: ToolDefinition.VISUALIZATION},
-            container_input_path=self.tool_definition.container_input_path,
-            input=self._create_container_input_dict(),
-            extra_directories=self.tool_definition.get_extra_directories()
-        )
-
-        self._django_docker_client.run(container)
-
+        # Pulls docker image if it doesn't exist yet, and launches container
+        # asynchronously
+        start_container.delay(self)
         return JsonResponse({Tool.TOOL_URL: self.get_relative_container_url()})
```

Comment:
Was it synchronous before? I worry a little about race conditions... are we sure it will be able to return a container_url no matter where in the launch sequence it is?

Reply:
It was before but now that we have the nice please wait page I think this is okay. 
A VisualizationTool will always be able to return its `get_relative_container_url` regardless of the container state.

Without this, one could hit a state where a large docker image hasn't be downloaded yet, and upon clicking `launch` the page would seemingly be frozen, while the pending request waits for the image download to complete.